filebeat/harvester/log.go

```diff
@@ -240,15 +236,9 @@ func (h *Harvester) getState() file.State {
 		return file.State{}
 	}
 
-	h.refreshState()
-	return h.state
-}
-
-// refreshState refreshes the values in State with the values from the harvester itself
-func (h *Harvester) refreshState() {
-	h.state.Source = h.path
-	h.state.Offset = h.getOffset()
+	// refreshes the values in State with the values from the harvester itself
 	h.state.FileStateOS = file.GetOSState(h.state.Fileinfo)
+	return h.state
```

Comment:
Yes, we should optimize this. But this was the case before already, it was just no visible very well.

In general the filestate of a Fileinfo should never change as long as the Fileinfo doesn't change? So we would have to read this only once?